internal/controller/nonadminbackup_controller_test.go

```diff
@@ -238,6 +248,14 @@ var _ = ginkgo.Describe("Test single reconciles of NonAdminBackup Reconcile func
 				}
 				gomega.Expect(k8sClient.Create(ctx, veleroBackup)).To(gomega.Succeed())
 			}
+
+			// We allow to have deleteNonAdminBackup and forceDeleteNonAdminBackup set to true at the same time
+			if scenario.nonAdminBackupSpec.ForceDeleteBackup {
```

Comment:
what you want to do with this? Add DeletionTimestamp to some tests?

Reply:
Some of the reconcile logic is based on the `DeletionTimestamp.IsZero()` so I want to ensure that object is pending deletion to test just part responsible for it (one function instead of entire switch part)

Reply:
here is example of failure of the test without this:
```shell
••••••••
------------------------------
• [FAILED] [0.014 seconds]
Test single reconciles of NonAdminBackup Reconcile function Reconcile triggered by NonAdminBackup Create/Update events and by Requeue [It] When triggered by NonAdminBackup forceDeleteNonAdmin spec field with Finalizer set, should delete NonAdminBackup and exit
./internal/controller/nonadminbackup_controller_test.go:526

  Timeline >>
  2024-11-13T17:06:54+01:00	DEBUG	NonAdminBackup Reconcile start
  2024-11-13T17:06:54+01:00	DEBUG	Executing force delete path
  2024-11-13T17:06:54+01:00	DEBUG	NonAdminBackup status unchanged during deletion
  2024-11-13T17:06:54+01:00	DEBUG	Marking NonAdminBackup for deletion	{"name": "nab-object-8"}
  [FAILED] in [It] - ./internal/controller/nonadminbackup_controller_test.go:266 @ 11/13/24 17:06:54.943
  << Timeline

  [FAILED] Expected
      <reconcile.Result>: {Requeue: true, RequeueAfter: 0}
  to equal
      <reconcile.Result>: {Requeue: false, RequeueAfter: 0}
  In [It] at: ./internal/controller/nonadminbackup_controller_test.go:266 @ 11/13/24 17:06:54.943
------------------------------
•••••


```

Reply:
but you are doing it based on nonAdmin object, not the test. I think this is wrong

correct approach would be creating flag like `addDeletionTimestamp` and passing as true to that test

Reply:
yes, however in this scenario we don't require another flag because there is no case where we will use this without force delete. Is to basically go into ForceDelete switch statement which is controlled by that `DeletionTimestamp.IsZero()` flag in the https://github.com/migtools/oadp-non-admin/pull/108/files#diff-98b8a647a66c83f5ee56f93b7530fb37a28301390936ecdb1fdaf38c031453c6R116

Reply:
Added `addNabDeletionTimestamp` as test flag.